src/main/java/se/sundsvall/parkingpermit/businesslogic/worker/AutomaticDenialDecisionTaskWorker.java

```diff
@@ -1,27 +1,94 @@
+import static se.sundsvall.parkingpermit.integration.casedata.mapper.CaseDataMapper.toLaw;
+import static se.sundsvall.parkingpermit.integration.casedata.mapper.CaseDataMapper.toStakeholder;
+
+import java.util.Objects;
+
 import org.camunda.bpm.client.spring.annotation.ExternalTaskSubscription;
 import org.camunda.bpm.client.task.ExternalTask;
 import org.camunda.bpm.client.task.ExternalTaskService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
+import org.zalando.problem.Problem;
+import org.zalando.problem.Status;
+
+import generated.se.sundsvall.casedata.ErrandDTO;
+import generated.se.sundsvall.casedata.StakeholderDTO;
+import se.sundsvall.parkingpermit.service.MessagingService;
 
 @Component
 @ExternalTaskSubscription("AutomaticDenialDecisionTask")
 public class AutomaticDenialDecisionTaskWorker extends AbstractWorker {
 	private static final Logger LOGGER = LoggerFactory.getLogger(AutomaticDenialDecisionTaskWorker.class);
+	private static final String PROCESS_ENGINE_FIRST_NAME = "Process";
+	private static final String PROCESS_ENGINE_LAST_NAME = "Engine";
+
+	@Autowired
+	private MessagingService messagingService;
 
 	@Override
 	public void execute(ExternalTask externalTask, ExternalTaskService externalTaskService) {
 		try {
-			LOGGER.info("Execute Worker for AutomaticDenialDecisionTask");
+			final var errand = getErrand(externalTask);
+			LOGGER.info("Executing automatic addition of dismissal to errand with id {}", errand.getId());
+
+			// PE needs to be added as stakeholder to the errand (if not already present) and store for later use when setting
+			// "decidedBy" on decision
+			final var stakeholder = ofNullable(errand.getStakeholders()).orElse(emptyList())
+				.stream()
+				.filter(AutomaticDenialDecisionTaskWorker::isProcessEngineStakeholder)
+				.findAny()
+				.orElseGet(() -> createProcessEngineStakeholder(errand));
+
+			final var props = messagingService.getProperties();
+			final var pdf = messagingService.renderPdf(errand);
+			final var decision = toDecision(FINAL, DISMISSAL, props.dismissalDescription())
+				.decidedBy(stakeholder)
+				.addLawItem(toLaw(props.lawHeading(), props.lawSfs(), props.lawChapter(), props.lawArticle()))
+				.addAttachmentsItem(toAttachment(BESLUT, props.filename(), "pdf", APPLICATION_PDF_VALUE, pdf));
+
+			caseDataClient.patchNewDecision(errand.getId(), decision);
 
 			externalTaskService.complete(externalTask);
 		} catch (Exception exception) {
-			LOGGER.error("Exception occurred in execution for task with id {} and businesskey {}", externalTask.getId(), externalTask.getBusinessKey());
+			LOGGER.error("Exception occurred in {} for task with id {} and businesskey {}", this.getClass().getSimpleName(), externalTask.getId(), externalTask.getBusinessKey(), exception);
```

Comment:
Is exception parameter necessary?

Reply:
I added it to see stacktrace where stuff went wrong when testing the flow. But it might be overkill to keep it, or could it be informational to see the row where the exception occurred (I don't think you'll see that if you don't provided the exception, right)?